### PR TITLE
Use eslint flat config

### PR DESCRIPTION
**Pull request by klebba**
Modernize config per https://eslint.org/docs/latest/use/configure/configuration-files-new

Address project rot:
- update dependencies; bump major version
- descope overbearing rules e.g. `'semi': [ 2, 'never' ]`
- descope babel & jsx rules
- publish package under `@Netflix` namespace
- address critical [security alerts](https://github.com/Netflix/eslint-config-netflix/security/dependabot)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+See ESLint docs on [flat configuration](https://eslint.org/docs/latest/use/configure/configuration-files-new)
+
+```sh
+npm install --save-dev @Netflix/eslint-config-netflix
+npm test
+```
+
+Philosophy:
+* Aim to enforce rules which reduce syntax related software bugs
+* Adopt rules that steer developers awaom from outdated language features (e.g. `no-var`)
```
Comment (theengineear) [inline README.md:12]: Spell-check `awaom`?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+* Avoid the temptation to enforce stylistic preferences; integrators should do this locally
```
Comment (theengineear) [inline README.md:13]: ❤️

**Review by theengineear — APPROVED** [review]
LGTM! Thanks for all the updates! Just a minor spellcheck — otherwise, all good.

**Comment by klebba** [comment]
Thanks for taking a look and catching the typo!

**Review by theengineear — APPROVED** [review]
Looks good!

**Pull request merged**